scripts/deploy.py

```diff
@@ -0,0 +1,497 @@
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout release"))
+    p.AddStep(RunCmdStep('git merge release', doc="Merge release into master"))
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(WriteAndCommitVersion('new-version', add_tag=True))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.AddStep(
+        RunCmdStep(
+            'git push --all origin', doc="Push release changes to github"))
+    p.AddStep(RunCmdStep('git push --tags origin', doc="Push tags to github"))
+    p.Run('new-release')
+
+
+@cli.command()
+@click.pass_context
+def cherrypick(ctx):
+    p = ctx.obj['pipeline']
+
+    CheckOwnIntegrityAndFetch(p)
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(SelectCommitToCherrypick(jump_on_end=3))
+    p.AddStep(CherryPick)
+    p.AddStep(RunJump(-2))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.Run('cherrypick')
+
+
+@cli.command()
+@click.pass_context
+def new_patch_release(ctx):
+    p = ctx.obj['pipeline']
+    CheckOwnIntegrityAndFetch(p)
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(ReadCurrentVerionFromFile)
+    p.AddStep(GetNextPatchVersion)
+    UpdateChangeLog(p)
+    p.AddStep(WriteAndCommitVersion('new-version', add_tag=True))
+    p.AddStep(
+        RunCmdStep(
+            'git push --all origin', doc="Push release changes to github"))
+    p.AddStep(RunCmdStep('git push --tags origin', doc="Push tags to github"))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.Run('new-patch')
+
+
+def CheckOwnIntegrityAndFetch(p):
+    p.AddStep(RunStoreFileSha(OWN_FILENAME))
+    p.AddStep(
+        RunCmdStep(
+            'git diff-index --quiet HEAD --',
+            doc="Check that git doesn't have uncommited changes"))
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(RunCmdStep('git pull', doc="Pull release from upstream"))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.AddStep(RunCmdStep('git pull', doc="Pull master from upstream"))
+    p.AddStep(CompareFileSha(OWN_FILENAME, require_equal=True))
+
+
+def UpdateChangeLog(p):
+    p.AddStep(SaveGitLog)
+    p.AddStep(PrependGitLogToChangeLog)
+    p.AddStep(RunStoreFileSha(CHANGELOG_PATH))
+    p.AddStep(
+        RunCmdStep(
+            '%s %s' % (os.getenv('EDITOR', 'vim'), CHANGELOG_PATH),
+            doc="Edit changelog"))
+    p.AddStep(CompareFileSha(CHANGELOG_PATH, require_equal=False))
+    p.AddStep(RunCmdStep('git add %s' % CHANGELOG_PATH))
+
+
+############################################################################
+# Pipeline machinery
+############################################################################
+
+
+def RunJump(where):
+    def f(ctx):
+        raise Jump(where)
+
+    if where > 0:
+        f.__doc__ = "Go %d steps forward" % where
```

Comment:
It's allowed in the style guide: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#310-strings

But I've replaced it in the entire file except one place where % does look more readable.

Reply:
Well the zen of python is that there should only be one way to do things, and string formatting is certainly one of the minor blemishes where they failed to hold to that ideal. If you're writing in a codebase that is not Python 2 compatible, such as this one, there's no reason really to use the outdated % formatting, and perhaps the Google style guide will reflect that in the future.

At any rate, f-strings (with `format()` as a supplement) are certainly the way of the future, but alas, it will be a few years before they can be assumed present.